src/main/scala/com/fulcrumgenomics/umi/ConsensusCaller.scala

```diff
@@ -20,448 +20,167 @@
-    for (baseIdx <- likelihoods.indices) { // for each base in the read
-      if (numReads(baseIdx) < this.options.minReads) {
-        consensusBases(baseIdx) = SequenceUtil.N
-        consensusQualities(baseIdx) = PhredScore.MinValue
-      }
-      else {
-        // get the sum of the likelihoods
-        // pick the base with the maximum posterior
-        val lls  = likelihoods(baseIdx)
-        val likelihoodSum   = LogProbability.or(lls)
-        val (maxLikelihood, maxLlIndex) = MathUtil.maxWithIndex(lls)
-        val maxPosterior    = LogProbability.normalizeByLogProbability(maxLikelihood, likelihoodSum)
-        val pConsensusError = LogProbability.not(maxPosterior) // convert to probability of the called consensus being wrong
-
-        // Masks a base if the phred score would be too low
-        consensusBases(baseIdx) = {
-          if (PhredScore.fromLogProbability(pConsensusError) < this.options.minConsensusBaseQuality) SequenceUtil.N
-          else DnaBasesUpperCase(maxLlIndex)
-        }
-
-        // Factor in the pre-UMI error rate.
-        // Pr(error) = Pr(any pre-UMI error AND correct consensus) + Pr(no pre-UMI error AND any error in consensus)
-        //               + Pr(pre-UMI error AND error in consensus, that do not give us the correct bases)
-        // The last term tries to capture the case where a pre-UMI error modifies the base (ex. A->C) but a sequencing
-        // error calls it the correct base (C->A).  Only 2/3 times will the two errors result in the incorrect base.
-        val p = probabilityOfErrorTwoTrials(this.options.errorRatePreUmiLn, pConsensusError)
-
-        // Cap the quality
-        consensusQualities(baseIdx) = PhredScore.cap(PhredScore.fromLogProbability(p))
-      }
-    }
-
-    ConsensusRead(consensusBases, consensusQualities)
-  }
-
   /**
-    * Adjusts the given base qualities.  The base qualities are first shifted by `baseQualityShift`, then capped using
-    *`maxBaseQuality`, and finally the `errorRatePostUmi` is incorporated.
-    *
-    * Implemented as a while loop with assignment into pre-created arrays as this function is called on every
-    * single input read and unfortunately Array[Double].map() causes boxing on all the values which is a
-    * significant performance drain.
+    * An inner class for tracking the likelihoods for the consensus for a single base.
     */
-  private[umi] def adjustBaseQualities(read: SourceRead): AdjustedRead = {
-    val len = read.length
-    val pErrors   = new Array[LogProbability](len)
-    val pCorrects = new Array[LogProbability](len)
-
-    val qs = read.quals
-    var i = 0
-    while (i < len) {
-      val q = qs(i)
-      val newQ = Math.min(this.options.maxBaseQuality, Math.max(PhredScore.MinValue, q - this.options.baseQualityShift))
-      pErrors(i)   = this.phredToLogProbError(newQ)
-      pCorrects(i) = this.phredToLogProbCorrect(newQ)
-      i += 1
+  class ConsensusBaseBuilder () {
```

Comment:
I had it as private, and then the outer class couldn't call it!  I'll kill the parens.  I think even though it's public the outer class is the only one that can construct it anyway.